src/model.ts

```diff
@@ -302,21 +303,25 @@ class ModelClass implements Model, ChangeItem {
     }
 
     const { type, ...fieldChanges } = fieldArgs;
-    const field = new Field(fieldChanges, MutationMode.Update);
+    const field = new Field(fieldChanges as FieldArgs, MutationMode.Update);
     this.listener.registerChange(field);
     return this;
   }
 
   addRelationalField(passedFieldArgs: any): Model {
     const fieldArgs = { ...passedFieldArgs };
     fieldArgs.modelApiId = this.args.apiId;
+
+    const fieldTypeUpper = fieldArgs.type?.toUpperCase();
+    const fieldModelUpper = fieldArgs.model?.toUpperCase();
```

Comment:
The strings were making me anxious considering `passedFieldArgs` is of type any 😅 . Also its nice to normalise string match checks.